### PR TITLE
modify some codes of pkg/kube

### PR DESCRIPTION
**Pull request by gy95**
1. fix some comment errors, comply with programming specifications
2. fix some log errors

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pkg/kube/inject/inject.go [hunk 1]
```diff
@@ -465,10 +464,10 @@ func injectionStatus(pod *corev1.Pod) *SidecarInjectionStatus {
 
 	// default case when injected pod has explicit status
 	var iStatus SidecarInjectionStatus
-	if err := json.Unmarshal(statusBytes, &iStatus); err == nil {
-		return &iStatus
+	if err := json.Unmarshal(statusBytes, &iStatus); err != nil {
```
Comment (ramaraochavali) [inline pkg/kube/inject/inject.go:467]: What is the intent of this change?

File: pkg/kube/secretcontroller/secretcontroller.go [hunk 2]
```diff
@@ -141,7 +141,7 @@ func (c *ClusterStore) Get(secretKey string, clusterID cluster.ID) *Cluster {
 	return c.remoteClusters[secretKey][clusterID]
 }
 
-// Get existing clusters registered for the given secret
+// GetExistingClustersFor get existing clusters registered for the given secret
```
Comment (ramaraochavali) [inline pkg/kube/secretcontroller/secretcontroller.go:144]: GetExistingClustersFor returns

File: pkg/kube/inject/inject.go (hunk 1 above)
Comment (gy95) [reply pkg/kube/inject/inject.go:467]: IMO, the go programming specification is first process error circumstance , then process normal circumstance, so do the change

File: pkg/kube/secretcontroller/secretcontroller.go (hunk 2 above)
Comment (gy95) [reply pkg/kube/secretcontroller/secretcontroller.go:144]: ok

File: pkg/kube/inject/validate.go [hunk 3]
```diff
@@ -163,7 +163,7 @@ func splitPorts(portsString string) []string {
 func parsePort(portStr string) (int, error) {
 	port, err := strconv.ParseUint(strings.TrimSpace(portStr), 10, 16)
 	if err != nil {
-		return 0, fmt.Errorf("failed parsing port '%d': %v", port, err)
+		return 0, fmt.Errorf("failed parsing port '%s': %v", portStr, err)
```
Comment (howardjohn) [inline pkg/kube/inject/validate.go:166]: What problem is this solving?

Comment (gy95) [reply pkg/kube/inject/validate.go:166]: > What problem is this solving?

if err is not nil, port will be always `0` , so we should return the origin `string` format, but not parsed `int` format

**Review by hzxuzhonghu — APPROVED** [review]
/ok-to-test

**Pull request merged**